src/librustc/ty/layout.rs

```diff
@@ -312,39 +314,36 @@ impl Align {
             }
             if bytes != 1 {
                 Err(format!("`{}` is not a power of 2", align))
-            } else if pow > 0x0f {
-                Err(format!("`{}` is too large", align))
```

Comment:
Should this struct care about the 31 bit limit? I've done the checking for that closer to the parsing. The previous limit here was due to the limitations of using just 1 nibble, but now it has no such limit.